### PR TITLE
[Python] Configuration Injection query 

### PR DESCRIPTION
**Pull request by Sim4n6**
Hi, 

This PR contains a **Configuration Injection** query with a qhelp file and test files.

Regards
@Sim4n6

### CODE REVIEW

File: python/ql/src/experimental/Security/CWE-074bis/ConfigInjection.ql [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  /**
+   * An argument to a call to the methods `set()` or `add_section()` on a configuration
+   * parser object is considered a Sink if it is not a an immutable literal like a boolean
+   * or a string.
+   */
+  override predicate isSink(DataFlow::Node sink) {
+    exists(MethodCallNode set, Node obj, ConfigParser configparser |
+      TaintTracking::localTaint(configparser.getACall(), obj) and
+      set.calls(obj, ["set", "add_section"]) and
```
Comment (yoff) [inline python/ql/src/experimental/Security/CWE-074bis/ConfigInjection.ql:110]: Could you not just use API graphs here?

Comment (Sim4n6) [reply python/ql/src/experimental/Security/CWE-074bis/ConfigInjection.ql]: Considered in https://github.com/github/codeql/pull/13640/commits/66560b9cb6f28bfd19123bd83dbe378e6bd31507 by using `configparser.getReturn().getAValueReachableFromSource()` to catch all Calls and class instantiations of the subclasses of `ConfigParser` and `RawConfigParser`.

**Comment by Sim4n6** [comment]
In https://github.com/github/codeql/pull/13640/commits/cab54b6398cd4fe2acec52d221800ae73c17323f I have added more symbols that could be injected to break the Config structure.

**Comment by RasmusWL** [comment]
The bounty for this PR was [rejected by Security Lab](https://github.com/github/securitylab/issues/768#issuecomment-1693537481). We agree with the overall assessment, and therefore we're also going to close this PR.

Thanks for your (sustained) interest in CodeQL, and I'm looking forward to seeing more impactful submission from you in the future :blush:

**Pull request closed without merging**